### ISSUE TITLE
ModSecurity not putting some transactions to audit log

### ISSUE THREAD

**Issue by PeterP55P** [coreruleset/coreruleset#2427]
### Describe the bug

We are facing a problem when in certain cases ModSecurity is not tracking the blocking in the audit log despite we have set it as a default action and the rule is not having any logging property set. In the error log, we can see only the rule which was triggered due to the anomaly score has been reached but nothing about the rules which actually counted the score. In some other cases, this information is visible.

In modsecurity.conf we have logging of rules enabled

`SecAuditLogParts ABCIJDFHKZ`

In crs-setup.conf we have default logging set to capture everything in both error and audit log.

```
SecDefaultAction "phase:1,log,auditlog,pass"
SecDefaultAction "phase:2,log,auditlog,pass"
```

We have also tried other combinations like using this setup

`SecDefaultAction "phase:2,pass, nolog,auditlog"`

But notning helped.

The only thing we can see in the log is this part from the error log and when we need more details we need to enable debug and redo the action. In the example below I have changed the IP addresses for security reasons.

``2022/03/09 08:32:25 [error] 1149#1149: *13832 [client 1.1.1.1] ModSecurity: Access denied with code 403 (phase 2). Matched "Operator `Ge' with parameter `5' against variable `TX:ANOMALY_SCORE' (Value: `15' ) [file "/etc/nginx/owasp-modsecurity-crs/rules/REQUEST-949-BLOCKING-EVALUATION.conf"] [line "139"] [id "949110"] [rev ""] [msg "Inbound Anomaly Score Exceeded (Total Score: 15)"] [data ""] [severity "2"] [ver "OWASP_CRS/3.4.0-dev"] [maturity "0"] [accuracy "0"] [tag "application-multi"] [tag "language-multi"] [tag "platform-multi"] [tag "attack-generic"] [hostname "10.1.0.1"] [uri "/favicon.ico"] [unique_id "1646811145"] [ref ""], client: 1.1.1.1, server: example.com, request: "GET /favicon.ico HTTP/1.1", host: "example.com", referrer: "http://example.com/?q=%22%3E%3Cscript%3Ealert(1)%3C/script%3E%27"``

We need to find a way how to make sure every attempt will be logged into the audit log.

Thanks in advance for any help.

Peter

### Your Environment

* CRS version (e.g., v3.4.0dev):
* Paranoia level setting: 2
* ModSecurity version (3.0.6):
* Web Server and version (e.g., nginx 1.20.2):
* Operating System and version: Oracle Linux 7.9

**Comment by PeterP55P** [comment]
Issue with audit log modsecurity 3.x . Not all transactions are captured in audit log.

**Comment by dune73** [comment]
We used to have a bug where an individual CRS rule would be silent. But that was long fixed. Which CRS version are you running?

Do you have everything in your error log, or also missing stuff?

The following should definitely report in all CRS versions in error log and audit log alike:
```
$ curl http://$HOST/?exec=bin/bash
```

**Comment by PeterP55P** [comment]
> We used to have a bug where an individual CRS rule would be silent. But that was long fixed. Which CRS version are you running?
> 
> Do you have everything in your error log, or also missing stuff?
> 
> The following should definitely report in all CRS versions in error log and audit log alike:
> 
> ```
> $ curl http://$HOST/?exec=bin/bash
> ```

Well, that is the problem. When I ran your sample curl command I'm in the same situation. The only thing which is captured is from the error log. Nothing is in the audit log.

`2022/03/11 09:19:26 [error] 32335#32335: *4573873 [client 1.1.1.1] ModSecurity: Access denied with code 403 (phase 2). Matched "Operator `Ge' with parameter `5' against variable `TX:ANOMALY_SCORE' (Value: `5' ) [file "/etc/nginx/owasp-modsecurity-crs/rules/REQUEST-949-BLOCKING-EVALUATION.conf"] [line "139"] [id "949110"] [rev ""] [msg "Inbound Anomaly Score Exceeded (Total Score: 5)"] [data ""] [severity "2"] [ver "OWASP_CRS/3.4.0-dev"] [maturity "0"] [accuracy "0"] [tag "application-multi"] [tag "language-multi"] [tag "platform-multi"] [tag "attack-generic"] [hostname "10.1.1.1"] [uri "/"] [unique_id "1646986766"] [ref ""], client: 1.1.1.1, server: example.com, request: "GET /?exec=bin/bash HTTP/1.1", host: "example.com"`

Our environment is described below. Apart from that, we have SecDefaultAction set to capture everything in both the error log and audit log.

    CRS version (v3.4.0dev):
    Paranoia level setting: 2
    ModSecurity version (3.0.6):
    Web Server and version (nginx 1.20.2):
    Operating System and version: Oracle Linux 7.9

**Comment by dune73** [comment]
There is something wrong with your setup and I can conclude with confidence it is not a CRS problem.

What troubles me is the fact that you have 949110 in the logs but not the rest. Very odd.

I suggest you turn to the ModSecurity repo for help with this, since it is a problem with ModSecurity - or the webserver.

**Comment by PeterP55P** [comment]
Thanks, I will try ModSecurity Git then.

**Comment by dune73** [comment]
Thank you. And sorry we can not really help you on the fly.
